cfgov/unprocessed/js/modules/ClearableInput.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// Required polyfills for IE9.
+if ( !Modernizr.classlist ) { require( '../modules/polyfill/class-list' ); } // eslint-disable-line no-undef, global-require, no-inline-comments, max-len
+
+// Required modules.
+var atomicCheckers = require( '../modules/util/atomic-checkers' );
+
+/**
+ * ClearableInput
+ * @class
+ *
+ * @classdesc Initializes a new ClearableInput molecule.
+ *
+ * @param {HTMLNode} element
+ *   The DOM element within which to search for the molecule.
+ * @returns {Object} A ClearableInput instance.
+ */
+function ClearableInput( element ) {
+  var BASE_CLASS = 'input-contains-label';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'ClearableInput' );
+  var _inputDom = _dom.querySelector( 'input' );
+  var _clearBtnDom = _dom.querySelector( '.' + BASE_CLASS + '_after__clear' );
+
+  /**
+   * @returns {Object} The ClearableInput instance.
+   */
+  function init() {
+    _clearBtnDom.addEventListener( 'mousedown', _clearClicked );
+    _inputDom.addEventListener( 'keyup', _inputTyped );
```

Comment:
Not sure how to get around this. We need it to respond immediately on the press of the delete key that removes everything, so we need to be checking whether each key press is the one to remove all content.